src/api/constants.ts

```diff
@@ -1 +1,46 @@
-export const API_URL: string = 'https://pokeapi.co/api/v2';
+export const API_URL = 'https://pokeapi.co/api/v2';
+
+export type ApiResponse = {
+  name: string;
+  url: string;
+};
+
+export type PokeApiResponse = {
+  count: number;
+  next: string | null;
+  previous: string | null;
+  results: ApiResponse[];
+};
+
+export type AbilityDetail = {
+  effect_changes: {
+    version_group: { name: string; url: string };
+    effect_entries: {
+      effect: string;
+      language: { name: string; url: string };
+    }[];
+  }[];
```

Comment:
описание типов лучше делать простым и понятным, чтобы взглянув, сразу было понятно что это за тип - массив, объект, примитив, с такой записью можно потеряться в скобках)
плюс тип вида { name: string; url: string } повторяется - сразу в отдельный тип значит
```suggestion
  type BaseType = {
    name: string; 
    url: string
  }
    
  type EffectEntries = {
    effect: string;
    language: BaseType []
    
  type EffectChanges= {
    version_group: BaseType ;
    effect_entries: EffectEntries[];
  }
```
названия типов для примера